packages/harmony/src/components/select/Select/Select.tsx

```diff
@@ -1,253 +1,210 @@
-                shadow='far'
-                onClick={(e) => e.stopPropagation()}
-              >
-                <Flex
-                  p='s'
-                  direction='column'
-                  alignItems='flex-start'
-                  role='listbox'
-                  aria-label={selectedLabel ?? props['aria-label']}
-                  aria-activedescendant={selectedLabel}
-                  w='100%'
-                  css={{
-                    maxHeight: popupMaxHeight,
-                    overflowY: 'auto'
-                  }}
-                  ref={scrollRef}
-                >
-                  <Flex direction='column' w='100%' gap='s'>
-                    {optionsLabel ? (
-                      <Box pt='s' ph='m'>
-                        <Text variant='label' size='xs'>
-                          {optionsLabel}
-                        </Text>
-                      </Box>
-                    ) : null}
-                    <SelectOptions
-                      activeValue={activeValue}
-                      options={filteredOptions}
-                      optionRefs={optionRefs}
-                      onChange={handleOptionSelect(handleChange, setIsOpen)}
-                    />
-                  </Flex>
-                </Flex>
-              </Paper>
-            </Popup>
-          )}
-        </SelectPopupKeyHandler>
-      )}
-    </SelectInput>
+      {(activeValue) =>
+        filteredOptions.length === 0 ? (
+          <Flex justifyContent='center'>
+            <Text variant='body' color='subdued' size='s'>
+              {messages.noMatches}
+            </Text>
+          </Flex>
+        ) : (
+          filteredOptions?.map((option, index) => (
+            <MenuItem
+              ref={(el) => {
+                if (optionRefs && optionRefs.current && el) {
+                  optionRefs.current[index] = el
+                }
+              }}
+              key={option.value}
+              variant='option'
+              {...option}
+              onChange={handleChange}
+              isActive={option.value === activeValue}
+            />
+          ))
+        )
+      }
+    </OptionKeyHandler>
+  )
+
+  return (
+    <Flex ref={anchorRef}>
+      <TextInput
+        ref={mergeRefs([ref, inputRef])}
+        value={inputValue}
+        onChange={(e) => setInputValue(e.target.value)}
+        onClick={handleClick}
+        aria-haspopup='listbox'
+        aria-expanded={isOpen}
+        autoComplete='off'
+        endIcon={Icon}
+        IconProps={{ onClick: handleClickIcon }}
+        elevateLabel={!!selectedOption}
+        hidePlaceholder={!!selectedOption}
+        {...other}
+      />
+      {/* Overlay selected option */}
+      {selectedOption && !inputValue ? (
+        <MenuItem
+          variant='option'
+          {...selectedOption}
+          onChange={() => {}}
+          css={{
+            position: 'absolute',
+            top: 26,
+            left: 4,
```

Comment:
do these relate to spacing values?

Reply:
they are fairly one-off, but i can use spacing values!